pkg/cloud/cloud.go

```diff
@@ -1,37 +1,429 @@
+		lustreConfiguration.SetMetadataConfiguration(metadataConfig)
+	} else {
+		metadataConfig := &fsx.FileCacheLustreMetadataConfiguration{StorageCapacity: aws.Int64(DefaultMetadataStorageCapacity)}
+		lustreConfiguration.SetMetadataConfiguration(metadataConfig)
+	}
+
+	if perUnitStorageThroughput, ok := configMap[LustreConfigOptionsPerUnitStorageThroughput]; ok {
+		throughput, err := strconv.ParseInt(perUnitStorageThroughput, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("invalid input for MetadataConfiguration PerUnitStorageThroughput: %s", err)
+		}
+		lustreConfiguration.SetPerUnitStorageThroughput(throughput)
+	} else {
+		lustreConfiguration.SetPerUnitStorageThroughput(DefaultPerUnitStorageThroughput)
+	}
+
+	if fileCacheOptions.WeeklyMaintenanceStartTime != "" {
+		lustreConfiguration.SetWeeklyMaintenanceStartTime(fileCacheOptions.WeeklyMaintenanceStartTime)
+	}
+
+	input.SetLustreConfiguration(lustreConfiguration)
 
-	metadata, err := NewMetadataService(svc)
+	if fileCacheOptions.KmsKeyId != "" {
+		input.SetKmsKeyId(fileCacheOptions.KmsKeyId)
+	}
+
+	if fileCacheOptions.CopyTagsToDataRepositoryAssociations {
+		input.SetCopyTagsToDataRepositoryAssociations(true)
+	}
+
+	klog.V(4).Infof("CreateFileCacheInput: ", input.GoString())
+	output, err := c.fsx.CreateFileCacheWithContext(ctx, input)
 	if err != nil {
-		return nil, fmt.Errorf("could not get metadata from AWS: %v", err)
+		if isIncompatibleParameter(err) {
+			return nil, ErrFcExistsDiffSize
+		}
+		return nil, fmt.Errorf("CreateFileCache failed: %v", err)
 	}
 
-	return &cloud{
-		metadata: metadata,
+	if output.FileCache.LustreConfiguration.MountName == nil {
+		return nil, fmt.Errorf("mountName returned in CreateFileCache is nil")
+	}
+	mountName := *output.FileCache.LustreConfiguration.MountName
+
+	perUnitStorageThroughput := int64(0)
+	if output.FileCache.LustreConfiguration.PerUnitStorageThroughput != nil {
+		perUnitStorageThroughput = *output.FileCache.LustreConfiguration.PerUnitStorageThroughput
+	}
+
+	return &FileCache{
+		FileCacheId:              *output.FileCache.FileCacheId,
+		CapacityGiB:              *output.FileCache.StorageCapacity,
+		DnsName:                  *output.FileCache.DNSName,
+		MountName:                mountName,
+		FileCacheType:            *output.FileCache.FileCacheType,
+		FileCacheTypeVersion:     *output.FileCache.FileCacheTypeVersion,
+		perUnitStorageThroughput: perUnitStorageThroughput,
 	}, nil
 }
 
-func (c *cloud) GetMetadata() MetadataService {
-	return c.metadata
+func (c *cloud) DeleteFileCache(ctx context.Context, fileCacheId string) (err error) {
+	input := &fsx.DeleteFileCacheInput{
+		FileCacheId: aws.String(fileCacheId),
+	}
+	if _, err = c.fsx.DeleteFileCacheWithContext(ctx, input); err != nil {
+		if isFileCacheNotFound(err) {
+			return ErrNotFound
+		}
+		return fmt.Errorf("DeleteFileCache failed: %v", err)
```

Comment:
Do we want to log the FC ID here as well or does that come with the error? Might be helpful to know